pywr/solvers/cython_glpk.pyx

```diff
@@ -632,15 +678,15 @@ cdef class CythonGLPKSolver(GLPKSolver):
         cdef double avail_volume
         cdef double t0
         cdef int col, row
-        cdef int* ind
-        cdef double* val
         cdef double lb
         cdef double ub
         cdef Timestep timestep
         cdef int status, simplex_ret
         cdef cross_domain_col
-        cdef list route
+        cdef list route, indptr_array
         cdef int node_id, indptr, nroutes
+        cdef int[::1] ind
```

Comment:
What does the `[::1]` syntax mean? Can't this just be `[:]`?

Reply:
It ensures the array is contiguous in memory. I think cython arrays are by default but I've seen `[::1]`used in a few examples where syntax like `&ind[0]` is used to get a pointer reference.

What I'm not sure about is whether setting a slice of the `agg_node.__agg_factor_data.ind` array to the `ind` variable in `_solve_scenario` involves the reallocation of some memory